### PR TITLE
Test: Add contract upgrade test

### PR DESCRIPTION
**Pull request by Dawn-githup**

### CODE REVIEW

File: contracts/test/UpgradeContract.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+const {expect} = require("chai");
+const {isGwMainnetV1} = require('../utils/network');
+const {ethers, upgrades} = require("hardhat");
+
+describe("ERC721MSHKUUPSTokenUpgrade", function () {
+    if (isGwMainnetV1()) {
+        return;
+    }
+
+    it("Deploy and call recursive functions", async () => {
```
Comment (Flouse) [inline contracts/test/UpgradeContract.js]: Please rename the test

File: contracts/contracts/ERC721MSHKUUPSTokenUpgrade.sol [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+contract ERC721MSHKUUPSTokenV2 is ERC721MSHKUUPSToken {
+
+    ///@dev increments the slices when called
+    function Decrement() external {
+        uint256 value = _tokenIdCounter;
+        require(value > 0, "Counter: decrement overflow");
+    unchecked {
+        _tokenIdCounter = value - 1;
+    }
+    }
```
Comment (Flouse) [inline contracts/contracts/ERC721MSHKUUPSTokenUpgrade.sol]: please format the code by the way.

**Comment by Flouse** [comment]
Re-running test:
- https://github.com/godwokenrises/godwoken-tests/actions/runs/3620068240/jobs/6140692170

**Review by Flouse — CHANGES_REQUESTED** [review]

File: contracts/test/UpgradeContract.js (hunk 1 above)
Comment (Dawn-githup) [reply contracts/test/UpgradeContract.js]: modified

**Comment by Dawn-githup** [comment]
> Re-running test:
> 
> * https://github.com/godwokenrises/godwoken-tests/actions/runs/3620068240/jobs/6140692170

This problem has been resolved

**Review by Flouse — APPROVED** [review]

**Pull request merged**